ext/RMagick/rmutil.c

```diff
@@ -1539,14 +1539,15 @@ rm_progress_monitor(
     VALUE rval;
     VALUE method, offset, span;
 
-// Default Ruby minimum stack size
-#define RUBY_VM_THREAD_MACHINE_STACK_SIZE_MIN (  16 * 1024 * sizeof(VALUE)) /*   64 KB or  128 KB */
-
-    // Check stack length manually instead of ruby_stack_check() for old Ruby.
-    if (ruby_stack_length(NULL) > RUBY_VM_THREAD_MACHINE_STACK_SIZE_MIN)
+    // Check running thread.
+    if (rm_current_thread_id() != rm_main_thread_id)
```

Comment:
Is there any way to send this message to the main thread?

Reply:
Unfortunately, I think there is no way to do that.

Reply:
If we can use private APIs, we might run monitor feature within GVL safely.

```c
    if (ruby_thread_has_gvl_p()) {
       // thread has a GVL.
       do_monitor_feature()
    }
    else if (ruby_native_thread_p()) {
       //  thread has no GVL. get
       rb_thread_call_with_gvl(do_monitor_feature, &argv);
    }
```
like, https://github.com/ruby/ruby/blob/f8555cad2d96c80fe7fc10012456e718cb81dbab/gc.c#L10271-L10278

Reply:
@Watson1978 would ruby core allow us to make them public APIs? For now I think what you've got here is good, though.